content/languages/python/authoring/index.md

```diff
@@ -0,0 +1,252 @@
+
+~~~if-not:python
+
+...text not visible in Python description...
+
+~~~
+```
+
+## Tasks and Requirements
+
+Some language constructs and features available in Python do not translate well into other languages, and should be avoided if possible:
+- Abuse of dynamic typing, e.g. mixed return types (_"Return the result, or the string 'Error' if no result can be found."_).
+
+Some kata should not be translated into Python, because it can be difficult to keep their initial idea:
+- The Python standard library is very rich and has many utilities available (e.g. `itertools`, combinatorics functions), so some nontrivial requirements in other languages become trivial in Python,
+- Python supports big integers natively, so kata that rely on the implementation of arbitrary precision integer arithmetic become trivial in Python.
+- The Codewars runner provides a set of preinstalled packages which are available for users solving the kata. They can be a real game-changer when it comes to the difficulty of a Python translation. For example, `numpy` can make many matrix manipulation kata much easier.
+
+## Coding
+
+
+### Code style
+
+Python code should stick to generally recognized Python conventions, with [PEP-8](https://www.python.org/dev/peps/pep-0008/) being most widely accepted.
+
+
+### Imports
+
+It is strongly recommended to import solution modules explicitly. Possible modules are:
+- `solution`, containing the user's submitted solution,
+- `preloaded`, containing preloaded code,
+- `codewars_test`, containing the [Codewars Python testing framework](/languages/python/codewars-test/).
+
+When any of these imports are missing, then, for compatibility reasons, it will be automatically added by Codewars preprocessor when the solution is built. However, relying on this behavior is discouraged, as it's considered deprecated and might change in the future.
+
+If a user solution is expected to use functions or classes from the `preloaded` module, it's recommended to add required imports to the "initial solution" code snippet.
+
+
+## Tests
+
+### Testing framework
+
+Python kata use the [Codewars Python testing framework](/languages/python/codewars-test/) to implement and execute tests. You should read its reference page to find out how to use `describe` and `it` blocks for [organization and grouping](/languages/python/codewars-test/#grouping-tests), what [assertions](/languages/python/codewars-test/#assertions) are available, etc.
+
+You should notice that the Python testing framework produces one test output entry per assertion, so the test output panel can get very noisy.
+
+### Random utilities
+
+Python has a rich [random library](https://docs.python.org/3.8/library/random.html), which can be used to easily generate random integers in requested ranges, generate floating-point numbers, or sample and shuffle collections. Functions available there allow for very convenient construction of various random input generators.
+
+Some useful functions include:
```

Comment:
```suggestion
Some useful functions include:
* [`random.choices(population[, ...], k=1)`](https://docs.python.org/3.8/library/random.html#random.choices) - extract `k` elements of the population (possibly outputing the same element several times).
```

Reply:
I specifically skipped `random.choices`, because I do not see how it can be useful in context of random tests. The fact that it can pick one element multiple times makes it IMO inferior to `random.sample`. Am I missing something here?

Reply:
about choices, it's about... choices. XD
Sorry... x)

that actually depends what you're building. If you're building data that can contain duplicated values, it's useful.

Reply:
I added a note on `choices`.